src/cookbook/navigation/set-up-universal-links.md

```diff
@@ -0,0 +1,202 @@
+scheme that is exclusive to Apple devices.
+
+Universal links lead users to apps instead of web browsers if
+the apps are installed on their Apple devices.
+
+## Prerequisite
+1. A mac with Flutter SDK and Xcode installed
+2. A web domain
+
+If you don’t own a web domain and still want to test the
+universal link, consider using [Firebase Hosting][] or
+[GitHub Pages][] as a temporary solution. These instructions
+use GitHub Pages.
+
+
+## 1. Write a Flutter application
+
+First, write a flutter application that can handle an incoming
+URL. Start by creating a new application with flutter create.
+
+```shell
+$ flutter create universallinks
+```
+
+This example uses [go_router][] to handle the routing. The
+go_router is a first party package maintained by Flutter
+that provides a simple API to handle complex routing scenarios.
+
+Add <strong>go_router</strong> to the <strong>pubspec.yaml</strong>:
+```yaml
+dependencies:
+  flutter:
+    sdk: flutter
+  go_router: ^6.0.0
+```
+
+In the <strong>main.dart</strong> file, create a <strong>GoRouter</strong>
+object to handle the routing:
+
+<?code-excerpt "lib/main.dart"?>
+```run-dartpad:theme-light:mode-flutter:run-true:width-100%:height-600px:split-60:ga_id-interactive_example
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+void main() => runApp(MaterialApp.router(routerConfig: router));
+
+/// The handles '/' and '/details'.
+final router = GoRouter(
+  routes: [
+    GoRoute(
+      path: '/',
+      builder: (_, __) => Scaffold(
+        appBar: AppBar(title: const Text('Home Screen')),
+      ),
+      routes: [
+        GoRoute(
+          path: 'details',
+          builder: (_, __) => Scaffold(
+            appBar: AppBar(title: const Text('Details Screen')),
+          ),
+        ),
+      ],
+    ),
+  ],
+);
+```
+
+
+
+## 2. Adjust iOS build settings
+
+Launch Xcode. Open the <strong>ios/Runner.xcworkspace</strong>
+file inside the project’s Flutter folder. Navigate to the
+<strong>info.Plist</strong> file:
+
+<noscript>
+  <img src="/assets/images/docs/cookbook/set-up-universal-links-info-plist.png" alt="Xcode info.Plist screenshot"/>
+</noscript>
+
+Add a row to the <strong>info.Plist</strong> with the key
+<strong>FlutterDeepLinkingEnabled</strong> and a
+<strong>Boolean</strong> value set to <strong>YES</strong>.
+
+<noscript>
+  <img src="/assets/images/docs/cookbook/set-up-universal-links-flutterdeeplinkingenabled.png" alt="flutter deeplinking enabled screenshot"/>
+</noscript>
+
+Next, navigate to <strong>Runner-> Sign & Signature -> + Capability -> Associated Domains</strong>
+
+<noscript>
+  <img src="/assets/images/docs/cookbook/set-up-universal-links-associated-domains.png" alt="Xcode associated domains screenshot"/>
+</noscript>
+
+In the <strong>Associated Domains</strong> section click +
+and enter <strong>applinks:\<web domain\></strong>.
+This recipe uses the domain name: chunhtai.github.io.
```

Comment:
A note to myself, replace `chunhtai.github.io` with more appropriate one

Reply:
@johnpryan I think this needs to be under https://flutter.github.io/samples/# somehow. Preferences?

Reply:
This is for github page. and the url to the AASA file has to be under
https://flutter.github.io/.well-known/apple-app-site-association.

I think we need a new account for this.